prov/tcp/src/tcpx_rma.c

```diff
@@ -72,6 +72,7 @@ static void tcpx_rma_read_send_entry_fill(struct tcpx_xfer_entry *send_entry,
 	send_entry->iov[0].iov_base = (void *) &send_entry->hdr;
 	send_entry->iov[0].iov_len = offset;
 	send_entry->iov_cnt = 1;
+	send_entry->flags = TCPX_INTERNAL_XFER;
```

Comment:
You know, I don't know if anyone can figure out what a function named "tcpx_rma_read_send_entry_fill" is actually supposed to do...

If I'm following the RMA read flow properly, tcp is queuing a send that initiates the RMA read request.  It also queues a receive for the reply.  So this is not really an internal message, but one generated for the user.  If the send fails, the RMA read operation has failed, and that failure should be reported back to the user.

Reply:
ok, than i interpretted the path incorrectly and believe the fix should then be to set `send_entry->context = msg->context;`

Reply:
Walking through the other parts of the code...

The receive (read response) is queued.  On any failure, the ep will be disabled, which will flush the queued receive.  That will generate a cancel error to rxm.  Discarding the send error can end up working since the receive will report an error.  Only the receive error will be CANCELED -- meaning the RMA read will be marked as canceled, even if the read request was the transfer that failed.

I'm not seeing a great way out of this predicament.  If the RMA read request is the active transfer when the socket send fails, ideally, we should report that failure code for the RMA operation.  If the RMA read request was not active (either the send completed or is still queued), then reporting CANCELED works.

It's almost like the receive should be marked internal to start with, and once the send completes, it should be marked internal and the receive marked external.